scarb/src/sources/registry.rs

```diff
@@ -120,28 +128,36 @@ impl<'c> Source for RegistrySource<'c> {
             .iter()
             // NOTE: We filter based on IndexRecords here, to avoid unnecessarily allocating
             //   PackageIds just to abandon them soon after.
+            // NOTE: Technically, RegistryClientCache may already have filtered the records,
+            //   but it is not required to do so, so we do it here again as a safety measure.
```

Comment:
I do not see a reason why it would not be required to do so since it gets full ManifestDependency now 🤔

Reply:
I would rather not mix this responsibility into client objects and cache. I want to treat cache as a layer over clients that has +/- the same usage contract. Only the cache layer does the filtering, clients themselves do not.